app/src/main/java/com/kylecombes/micrepeater/MainActivity.java

```diff
@@ -68,20 +69,14 @@ public void bindingViews() {
         bluetoothStatusTV = findViewById(R.id.textView_main_bluetoothStatus);
         startButton = findViewById(R.id.button_main_start);
         stopButton = findViewById(R.id.button_main_stop);
-        firebaseSwitch = findViewById(R.id.firebase_switch);
+        settingButton = findViewById(R.id.button_setting);
         versionNumber = findViewById(R.id.version);
         versionNumber.setText(getResources().getString(R.string.version, BuildConfig.VERSION_CODE));
     }
 
-    public void setUpFirebaseAnalytics() {
-        firebaseSwitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
-            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
-                firebaseAnalyticsOn = isChecked;
-            }
-        });
-
-        // Log events and crashes
-        if (firebaseAnalyticsOn) mFirebaseAnalytics = FirebaseAnalytics.getInstance(this);
+    public static void activateFirebase(){
+        firebaseAnalyticsOn = !firebaseAnalyticsOn;
```

Comment:
Flipping the boolean would work, but I don't suggest doing this because we don't know what the current boolean value is. It's better to explicitly have a boolean variable `isChecked` which can be obtained from the Switch function `onCheckedChanged(CompoundButton buttonView, boolean isChecked)`.

Another thing is the function name might be confusing because the function both activates and deactivates Firebase at the same time. 
- If you don't want to use the `isChecked` variable I mentioned above (which is fine), you can name the function `toggleFirebase()` or something. 
- If you use the `isChecked` variable, I would suggest naming it `void enableFirebase(boolean isChecked)` -> so when using it, you would do something like `enableFirebase(true);`